internal/controller/clientprofilemapping_controller.go

```diff
@@ -18,45 +18,184 @@ package controller
-		For(&csiv1alpha1.ClientProfileMapping{}).
+		For(&csiv1a1.ClientProfileMapping{}).
+		// TODO: This watch is probalmatic, it will trigger a reconicle for each
+		// mapping resource that exists in the configmap namespace whenever the configmap
+		// is changed. This is unnececary as this reconile build its desired state based
+		// on all mapping resources withing a single reconicle
+		Owns(
+			&corev1.ConfigMap{},
+			builder.MatchEveryOwner,
+			builder.WithPredicates(
+				utils.NamePredicate(utils.CsiConfigVolume.Name),
+				genChangedPredicate,
+			),
+		).
 		Complete(r)
 }
+
+func (r *ClientProfileMappingReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := ctrllog.FromContext(ctx)
+	log.Info("Starting reconcile iteration for ClientProfileMapping", "req", req)
+
+	reconcileHandler := ClientProfileMappingReconcile{}
+	reconcileHandler.ClientProfileMappingReconciler = *r
+	reconcileHandler.ctx = ctx
+	reconcileHandler.log = log
+	reconcileHandler.req = req
+
+	err := reconcileHandler.reconcile()
+	if err != nil {
+		log.Error(err, "ClientProfileMapping reconciliation failed")
+	} else {
+		log.Info("ClientProfileMapping reconciliation completed successfully")
+	}
+	return ctrl.Result{}, err
+}
+
+func (r *ClientProfileMappingReconcile) reconcile() error {
+	// This controller behave a differently then other controller. Becaue of the lack of uniqueness of the mapping
+	// inforation between 2 or more mapping resources, there is no way to update the csi mapping information without
+	// building it from the gournd up on every reconicle. To do that we ignore the specific of the subject resource
+	// and instead load all mapping resources within the namespace.
+	if err := r.List(r.ctx, &r.clientProfileMappingList, client.InNamespace(r.req.Namespace)); err != nil {
+		r.log.Error(err, "Failed listing ClientProfileMapping CRs in namspace", "namespace", r.req.Namespace)
+		return err
+	}
+
+	if err := r.reconcileCephCsiBlockPoolMapping(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *ClientProfileMappingReconcile) reconcileCephCsiBlockPoolMapping() error {
+	csiConfigMap := corev1.ConfigMap{}
+	csiConfigMap.Name = utils.CsiConfigVolume.Name
+	csiConfigMap.Namespace = r.req.Namespace
+
+	log := r.log.WithValues("csiConfigMapName", csiConfigMap.Name)
+	log.Info("Reconciling Ceph CSI Cluster Info")
+
+	_, err := ctrlutil.CreateOrUpdate(r.ctx, r.Client, &csiConfigMap, func() error {
+		var owner *csiv1a1.ClientProfileMapping
+		for i := range r.clientProfileMappingList.Items {
+			item := &r.clientProfileMappingList.Items[i]
+			if item.Name == r.req.Name && item.Namespace == r.req.Namespace {
+				owner = item
+				break
+			}
+
+		}
+		if owner != nil {
+			if _, err := utils.ToggleOwnerReference(true, &csiConfigMap, owner, r.Scheme); err != nil {
+				log.Error(err, "Failed toggling owner reference for Ceph CSI config map")
+				return err
+			}
+		}
+
+		type MappingKey [2]string
```

Comment:
`type` or `var`? title case mandatory?

Reply:
This is a local type definition! title case is not mandatory.

Reply:
yes, is there any reason declaring it as a `type` rather than `var`?

Reply:
Fixed the letter casing

Reply:
> yes, is there any reason declaring it as a `type` rather than `var`?

Leela the `mappingKey` is a type definition, if I declare it as var it will no longer be a type definition.
This type is being utilized to declare values that are used as keys to a map

Reply:
Thx, pls resolve.